content-based-message-routing/README.md

```diff
@@ -0,0 +1,476 @@
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(ReservationRequest payload) 
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        
+    }
+```
+
+7. Implement the logic
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function post categories/[string category]/reserve(ReservationRequest payload)
+            returns ReservationResponse|http:NotFound|http:BadRequest|http:InternalServerError {
+        http:Client? hospitalEP = ();
+        ReservationRequest {hospital_id, patient, ...reservationRequest} = payload;
+        match hospital_id {
+            "grandoaks" => {
+                log:printInfo("Routed to Grand Oak Community Hospital");
+                hospitalEP = grandOaksEP;
+            }
+            "clemency" => {
+                log:printInfo("Routed to Clemency Medical Center");
+                hospitalEP = clemencyEP;
+            }
+            "pinevalley" => {
+                log:printInfo("Routed to Pine Valley Community Hospital");
+                hospitalEP = pineValleyEP;
+            }
+        }
+
+        if hospitalEP is () {
+            log:printError(string `Routed to none. Hospital not found: ${hospital_id}`);
+            return <http:NotFound>{body: string `Hospital not found: ${hospital_id}`};
+        }
+
+        ReservationResponse|http:ClientError resp = hospitalEP->/[category]/reserve.post({patient, ...reservationRequest});
+
+        if resp is ReservationResponse {
+            log:printDebug("Reservation request successful",
+                            name = patient.name,
+                            appointmentNumber = resp.appointmentNumber);
+            return resp;
+        }
+
+        log:printError("Reservation request failed", resp);
+        if resp is http:ClientRequestError {
+            return <http:NotFound>{body: "Unknown hospital, doctor or category"};
+        }
+
+        return <http:InternalServerError>{body: resp.message()};
+    }
+}
+```
+
+- Initialize `hospitalEP` to a null value. Later, this variable is assigned to the relevant client based on the `hospital_id` value.
```

Comment:
null is for JSON contexts. Use nil instead. Also mention that the type is a union of `http:Client` and nil.

Reply:
The relevant client is assigned to this variable. Currently reads the other way around.